beets/autotag/__init__.py

```diff
@@ -166,7 +166,38 @@ def ensure_first_value(single_field: str, list_field: str) -> None:
         elif list_val:
             setattr(m, single_field, list_val[0])
 
+    def migrate_legacy_genres() -> None:
+        """Migrate comma-separated genre strings to genres list.
+
+        For users upgrading from previous versions, their genre field may
+        contain comma-separated values (e.g., "Rock, Alternative, Indie").
+        This migration splits those values into the genres list on first access,
+        avoiding the need to reimport the entire library.
+        """
+        genre_val = getattr(m, "genre", "")
+        genres_val = getattr(m, "genres", [])
+
+        # Only migrate if genres list is empty and genre contains separators
+        if not genres_val and genre_val:
+            # Try common separators used by lastgenre and other tools
+            for separator in [", ", "; ", " / "]:
+                if separator in genre_val:
+                    # Split and clean the genre string
+                    split_genres = [
+                        g.strip()
+                        for g in genre_val.split(separator)
+                        if g.strip()
+                    ]
+                    if len(split_genres) > 1:
+                        # Found a valid split - populate genres list
+                        setattr(m, "genres", split_genres)
+                        # Clear genre so ensure_first_value sets it correctly
+                        setattr(m, "genre", "")
+                        break
+
     ensure_first_value("albumtype", "albumtypes")
+    migrate_legacy_genres()
```

Comment:
This will only get called _during autotagging process_ (or using `migrate` command).

In order to switch from `genre` to `genres` for good, we want to migrate the values _at the same time **when the field is being created** in the database_.

That is, we want to call this function from within `Database._create_table` method after creating the new `genres` column. 

It means that the migration will be performed immediately when `beet` command is run for the first time. Users should be given no option here - migration _must_ take place. This means we do not need the `migrate` command.

That said, we should make sure that we have appropriate progress reporting - the migration will **write** every file that contains non-empty genre.